sunpy/map/map_factory.py

```diff
@@ -95,7 +97,16 @@ def _read_file(self, fname, **kwargs):
         # This can be removed once read_file supports pathlib.Path
         log.debug(f'Reading {fname}')
         try:
-            pairs = read_file(os.fspath(fname), **kwargs)
+            if str(fname).endswith("asdf"):
+                if minversion(asdf, "3.1.0"):
+                    _NO_MEMMAP_KWARGS = {"memmap": False, "lazy_load": False}
```

Comment:
I did that to avoid segmentation fault

Reply:
What versions of asdf segfault?

Reply:
it was version 2.13.0 i am not sure was this due to older version of asdf only or was it due to older version of numpy too

Reply:
i think the seg fault issue only persists with older version of asdf

Reply:
so i think for other version we can set memap as true

Reply:
I don't think this should be memory mapping any arrays. It closes the file after the map is generated (but the arrays are still in memory). The segfault is avoided in asdf>=3.1.0 only because it gives up trying to close the file (similar to numpy.memmap) since it sees a memory mapped array.

Reply:
Neither option fits the usage here (where the file is closed before the contents are freed).

If "performance = execution time" memory mapping is almost always worse in terms of array access time compared to loading an array in memory.

What is the performance issue?

Reply:
i meant execution time like this might increase memory usage